inc/search.class.php

```diff
@@ -4633,10 +4633,12 @@ static function giveItem($itemtype, $ID, array $data, $num, $meta = 0,
 
             case 'glpi_projectstates.name':
                $out = '';
-               $query = "SELECT `color`
-                         FROM `glpi_projectstates`
-                         WHERE `name` = '".$data[$num][0]['name']."'";
-               foreach ($DB->request($query) as $color) {
+               $color_iterator = $DB->request([
+                  'SELECT' => ['color'],
+                  'FROM'   => 'glpi_projectstates',
+                  'WHERE'  => ['name' => $DB->escape($data[$num][0]['name'])]
```

Comment:
is the escape really needed with iterator usage ?

Reply:
Yes, unfortunately... Data are """"magically"""" escaped when caming from a web request; but `$data` seems to come from a DB query; so nothing is escaped.